crypto/x509v3/v3_ncons.c

```diff
@@ -297,47 +297,143 @@ int NAME_CONSTRAINTS_check(X509 *x, NAME_CONSTRAINTS *nc)
 
 }
 
+#define CN2DNSID_NOMEM   (-2)
+#define CN2DNSID_BADNAME (-1)
+
+static int cn2dnsid(ASN1_STRING *cn, unsigned char **dnsid)
```

Comment:
I'm not at all keen on overloading the return value of functions. This returns a length (which should really be a size_t) or error codes (which are negative), and tries to merge the two concepts into an int. This causes real problems with lengths/sizes of things being cast from one type to another and then back again. We should avoid that.

Reply:
I could separate out the return value from the length.  Note that given that the output is the result of a conversion via `ASN1_STRING_to_UTF8()` we're still stuck with a signed `int` length. If we ever support size_t lengths for ASN1_STRING types, then this code would be able to support that if refactored to separate the length from success/failure status.  Would like to see that done now, even though we don't yet get any benefit?

Reply:
Personally I would prefer the length to be separated. I do think we should be looking to sort out the size_t/int mess at some point (we pretty much have in libssl, except where it touches libcrypto). So I'd rather not dig the hole any deeper. However, it's not in our coding style, so I won't insist on it.